client/state/sites/actions.js

```diff
@@ -158,7 +158,22 @@ export function requestSite( siteFragment ) {
 			.then( ( site ) => {
 				// If we can't manage the site, don't add it to state.
 				if ( site && site.capabilities ) {
-					dispatch( receiveSite( omit( site, '_headers' ) ) );
+					const state = getState();
+					const wasAtomic = state?.sites?.items?.[ siteFragment ]?.options?.is_wpcom_atomic;
+					const isAtomic = site?.options?.is_wpcom_atomic;
+					const wasAdmin = state?.currentUser?.capabilities?.[ siteFragment ]?.manage_options;
+					const isAdmin = site?.capabilities?.manage_options;
+
+					/*
+					 * Capabilities are not immediately propagated to the Atomic site
+					 * after transfer, so let's hold off updating the state until the
+					 * endpoint returns accurate data.
+					 */
+					if ( ! wasAtomic && isAtomic && wasAdmin && ! isAdmin ) {
```

Comment:
I understand that the `wasAdmin && ! isAdmin` bit is to check if the capabilites are outdated? If so, we could rename it to `areCapabilitesOutdated` or something along the lines.

Reply:
https://github.com/Automattic/wp-calypso/pull/95220/commits/84442b34bbcffeb19254d4665c52dc73a73ba65e